ambry-router/src/main/java/com.github.ambry.router/DeleteManager.java

```diff
@@ -100,10 +100,11 @@ void submitDeleteBlobOperation(String blobIdString, FutureResult<Void> futureRes
     try {
       BlobId blobId = RouterUtils.getBlobIdFromString(blobIdString, clusterMap);
       DeleteOperation deleteOperation =
-          new DeleteOperation(routerConfig, responseHandler, blobId, futureResult, callback, time);
+          new DeleteOperation(routerConfig, responseHandler, blobId, futureResult, callback, time, routerMetrics);
       deleteOperations.add(deleteOperation);
     } catch (RouterException e) {
       routerMetrics.operationDequeuingRate.mark();
+      routerMetrics.countError(e, NonBlockingRouter.RouterOperationType.DeleteBlob);
```

Comment:
1. Removed the logic to update metrics based on the type of an operation.
2. Removed most of the helper methods in `NonBlockingRouterMetrics`, to simply metrics updating at the callers. 
3. The only helper method kept is `countError(Exception e)`, because the switching branches are needed.